nautobot_circuit_maintenance/handle_notifications/handler.py

```diff
@@ -163,21 +163,20 @@ def create_or_update_circuit_maintenance(
     maintenance_id = f"{raw_entry.provider.slug}-{parser_maintenance.maintenance_id}"
     try:
         circuit_maintenance_entry = CircuitMaintenance.objects.get(name=maintenance_id)
-        # Using the RawNotification.date as the reference to sort because it's the one that takes into account the
-        # source receving time. The ParsedNotification.date stores the date when the RawNotification was parsed and the
-        # ParsedNotification was created.
+        # Using the RawNotification.stamp as the reference to sort because it's the one that takes into account the
+        # source receving time.
         last_parsed_notification = (
-            circuit_maintenance_entry.parsednotification_set.order_by("raw_notification__date").reverse().last()
+            circuit_maintenance_entry.parsednotification_set.order_by("raw_notification__stamp").reverse().last()
         )
 
         # If the notification is older than the latest one used to update the CircuitMaintenance, we skip updating it
         parser_maintenance_datetime = datetime.datetime.fromtimestamp(
             parser_maintenance.stamp, tz=datetime.timezone.utc
         )
-        if last_parsed_notification and last_parsed_notification.date > parser_maintenance_datetime:
+        if last_parsed_notification and last_parsed_notification.last_updated > parser_maintenance_datetime:
             logger.log_debug(
                 f"Not updating CircuitMaintenance {maintenance_id} because the notification is from "
-                f"{parser_maintenance_datetime}, older than the most recent notification from {last_parsed_notification.date}."
+                f"{parser_maintenance_datetime}, older than the most recent notification from {last_parsed_notification.created}."
```

Comment:
Should this be using `last_updated` rather than `created` for consistency with the actual logic above?

Reply:
totally